Source/Core/Common/Config/Layer.h

```diff
@@ -129,13 +129,13 @@ class Layer
     Set(location, ValueToString(value));
   }
 
-  void Set(const ConfigLocation& location, const std::string& new_value)
+  void Set(const ConfigLocation& location, std::string new_value)
   {
-    std::optional<std::string>& current_value = m_map[location];
-    if (current_value == new_value)
+    auto it = m_map.find(location);
+    if (it != m_map.end() && it->second == new_value)
       return;
     m_is_dirty = true;
-    current_value = new_value;
+    m_map[location] = std::move(new_value);
```

Comment:
```suggestion
    m_map.insert_or_assign(location, std::move(new_value));
```

performs the same behavior without default constructing a temporary std::string if it doesn't exist in the map.

Reply:
TIL! I wasn't aware of this, great :)